design/ingress-status-loadbalancer.md

```diff
@@ -0,0 +1,101 @@
+# Ingress Status Loadbalancer Support
+
+Status: _Draft_
+
+This proposal describes how to add support for the Ingress.Status.LoadBalancer field to Contour.
+
+## Goals
+
+- Populate the status.loadbalancer field in all networking.k8s.io/v1beta.Ingress objects managed by Contour.
+- Contour pass the Ingress compliance tests for networking.k8s.io/v1beta.Ingress objects.
+
+## Non Goals
+
+- Similar functionality for IngressRoute (deprecated) and HTTPProxy are out of scope.
+
+## Background
+
+The Kubernetes Ingress object provides a field on the object's status to record address details for how access the ingresses virtualhosts from outside the cluster.
+This is known as the status loadbalancer information after the name of the fields on the Ingress Object.
+
+Details from the Kubernetes spec [are here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#loadbalancerstatus-v1-core).
+
+## High-Level Design
+
+There are two components to this solution
+
+1. Discoverying the loadbalancer information
+2. Writing the loadbalacner information to the ingress objects.
+
+The former, the discovery will need to support 
+
+1. Hard coded IP/address details (actually a list, and it can be an address or an ip)
+2. Discoverying information from the service document.
+
+### Hard coded load balancer addresses
+
+While the hard coded approach sounds simpler, and is, it will likely not be the common use case, nor will it be the default.
+This is for several reasons.
+
+- The hard coded address model implies either the cluster is running on prem, or using host networking.
```

Comment:
This should probably be 'hard coded *name*'.

Additionally, it doesn't have to be host networking, it can be done with a HostPort and an external load balancer. The important part here is that name-based hard coding implies the use of an external load-balancer that can front the Envoy service. It might be on-prem, host networking, in AWS, or whatever. But the most common way to get a stable name is to use an external load balancer.

You could also front a pool of machines with a straight domain name, I suppose, but this would involve having Envoy listen on 80 and 443 and for those machines to have internet-reachable IPs.